src/helpers/api.ts

```diff
@@ -76,3 +76,40 @@ export async function fetchNetworkDifficulty(): Promise<NetworkDifficultyEstimat
     const response = await fetch(`${apiUrl}/network-difficulty`);
     return await response.json();
 }
+
+export interface SingleBlockData {
+    block: Block;
+}
+
+export async function fetchSingleBlock(blockId: Block): Promise<SingleBlockData> {
```

Comment:
The blockId is a number or string. Get rid of SingleBlockData, it is a BlockEntity.